tests/lib/utils/ember-test.js

```diff
@@ -192,25 +320,128 @@ describe('isEmberRoute', () => {
   describe("should check if it's an Ember Route", () => {
     it('should detect Route when using Ember.Route', () => {
       const node = parse('Ember.Route.extend()');
-      expect(emberUtils.isEmberRoute(node)).toBeTruthy();
+      const context = { getFilename() {} };
+      expect(emberUtils.isEmberRoute(context, node)).toBeTruthy();
     });
 
     it('should detect Route when using local module Route', () => {
       const node = parse('Route.extend()');
-      expect(emberUtils.isEmberRoute(node)).toBeTruthy();
+      const context = { getFilename() {} };
+      expect(emberUtils.isEmberRoute(context, node)).toBeTruthy();
+    });
+
+    it('should detect Route when using native classes', () => {
+      const code = babelEslint.parse(`
+        import Route from '@ember/routing/route';
+        class MyRoute extends Route {}
+      `);
+      const node = code.body[1];
+      const context = {
+        getFilename() {},
+        getAncestors() {
+          return [code, ...code.body];
+        },
+      };
+      expect(emberUtils.isEmberRoute(context, node)).toBeTruthy();
+    });
+
+    it('shouldnt detect Route when using native classes if the import path is incorrect', () => {
+      const code = babelEslint.parse(`
```

Comment:
I wrote a little test helper around creating a fake Context for testing (I think it might have been merged after this PR was created)

https://github.com/ember-cli/eslint-plugin-ember/blob/61010f3485723f7ecd829e2ac68486206a76ef2c/tests/lib/utils/utils/get-source-module-name-for-identifier-test.js#L10

Would it be helpful to put that in a shared location and use it between both tests instead?

Reply:
Ooh yes! Looks great, will do that here. Thank you!